assets/translate.js

```diff
@@ -127,8 +132,11 @@ $( window ).on( 'load', function () {
 				// Go through all fields and construct the request parameters.
 				$( '.translation-fields .oo-ui-fieldLayout' ).each( function () {
 					var fieldLayout = OO.ui.infuse( $( this ) ),
-						tspanId = fieldLayout.getField().data[ 'tspan-id' ];
-					requestParams[ tspanId ] = fieldLayout.getField().getValue();
+						tspanId = fieldLayout.getField().data[ 'tspan-id' ],
+						text = fieldLayout.getField().getValue(),
+						textChanged = text !== '' && text !== appConfig.translations[ tspanId ][ targetLangWidget.getValue() ];
+					requestParams[ tspanId ] = text;
+					canUpload = canUpload || ( textChanged && typeof appConfig.user === 'string' );
```

Comment:
If we're only using the new `appConfig.user` to check for logged-in status, I reckon it'd be clearer to just call it `appConfig.loggedIn` and make it boolean. As it is, this doesn't work always because `appConfig.user` is an object.